### PR TITLE
Unity Testsuite

### PR DESCRIPTION
**Pull request by jdetter**
## Description of Changes
*Describe what has been changed, any new features or bug fixes*

- This PR introduces a testsuite which runs in Unity. I have started with the [SpacetimeDBCircleGame](https://github.com/clockworklabs/SpacetimeDBCircleGame) but I think we should add more small example games in the future. Right now it just spawns in a circle, eats some food and verifies the decay logic is working correctly. I've also written some reconnection tests but they don't work because reconnections are currently broken. There are also one-off tests but those don't work either because they also require reconnections to be working.

Update: reconnections have been fixed via https://github.com/clockworklabs/com.clockworklabs.spacetimedbsdk/pull/168 . In that PR I have enabled all tests!

I've used the built-in unity testsuite framework to achieve this, along with the UnityCI tool from GameCI. The documentation for this docker container can be found here: https://game.ci/docs/github/getting-started/

Rendered README for our testsuite which includes local testing instructions: https://github.com/clockworklabs/com.clockworklabs.spacetimedbsdk/blob/jdetter/unity-testsuite/unity-tests~/README.md

## API

 - [ ] This is an API breaking change to the SDK

*If the API is breaking, please state below what will break*

Not breaking

## Requires SpacetimeDB PRs
*List any PRs here that are required for this SDK change to work*

None

## Testsuite

SpacetimeDB branch name: 0935b7346b825b8cbb9f36d9ed256136b73b5f0b

## Testing
*Write instructions for a test that you performed for this PR*

- [ ] Describe a test for this PR that you have completed

### CODE REVIEW

**Comment by jdetter** [comment]
Ingvar suggested we add a check which verifies that the client bindings on the branch being tested are up-to-date, I agree with this and I've added the check.

**Comment by cloutiertyler** [comment]
@jdetter I am going to approve this, however, I am also going to schedule a meeting with @RReverser @SteveBoytsun @lcodes @Centril with the purpose of having you explain in person how it works, so that we can hand it off to others with them being able to understand the tricky problems of setting this up.

**Comment by cloutiertyler** [comment]
Should merge after this: https://github.com/clockworklabs/com.clockworklabs.spacetimedbsdk/pull/177

**Comment by Centril** [comment]
Rebased this atop of a rebased `jdetter/unity-testsuite-import` which was rebased atop of staging (https://github.com/clockworklabs/com.clockworklabs.spacetimedbsdk/pull/177#issuecomment-2446883581) and changed back the merge-into target branch to clarify the diff for review purposes.

Also changed the test-against stdb commit to `0935b7346b825b8cbb9f36d9ed256136b73b5f0b`, currently the `HEAD` of `master`.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -10,3 +10,13 @@
 
 ## Requires SpacetimeDB PRs
 *List any PRs here that are required for this SDK change to work*
+
+## Testsuite
+*If you would like to run against a specific SpacetimeDB branch in the testsuite, specify that here. This can be a branch name or a link to a PR.*
+
+SpacetimeDB branch name: master
```
Comment (jdetter) [inline .github/pull_request_template.md:17]: let's call this a ref instead

File: .github/workflows/unity-test.yml [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+      - uses: actions/cache@v3
+        with:
+          path: unity-tests~/client/Library
+          key: Library-SpacetimeDBUnityTestsuite-Linux-x86
+          restore-keys: |
+            Library-SpacetimeDBUnityTestsuite-
+            Library-
+
+      - name: Set up Unity
```
Comment (jdetter) [inline .github/workflows/unity-test.yml:107]: add this link: https://game.ci/docs/github/getting-started/

**Comment by jdetter** [comment]
closed in favor of https://github.com/clockworklabs/com.clockworklabs.spacetimedbsdk/pull/186

**Pull request closed without merging**